fastplotlib/graphics/selectors/_rectangle.py

```diff
@@ -0,0 +1,511 @@
+import warnings
+from numbers import Real
+from typing import *
+import numpy as np
+
+import pygfx
+from .._collection_base import GraphicCollection
+
+from .._base import Graphic
+from .._features import RectangleSelectionFeature
+from ._base_selector import BaseSelector
+
+
+class RectangleSelector(BaseSelector):
+    @property
+    def parent(self) -> Graphic | None:
+        """Graphic that selector is associated with."""
+        return self._parent
+
+    @property
+    def selection(self) -> Sequence[float] | List[Sequence[float]]:
+        """
+        (xmin, xmax, ymin, ymax) of the rectangle selection
+        """
+        return self._selection.value
+
+    @selection.setter
+    def selection(self, selection: Sequence[float]):
+        # set (xmin, xmax, ymin, ymax) of the selector in data space
+        graphic = self._parent
+
+        if isinstance(graphic, GraphicCollection):
+            pass
+
+        self._selection.set_value(self, selection)
+
+    @property
+    def limits(self) -> Tuple[float, float, float, float]:
+        """Return the limits of the selector."""
+        return self._limits
+
+    @limits.setter
+    def limits(self, values: Tuple[float, float, float, float]):
+        if len(values) != 4 or not all(map(lambda v: isinstance(v, Real), values)):
+            raise TypeError("limits must be an iterable of two numeric values")
+        self._limits = tuple(
+            map(round, values)
+        )  # if values are close to zero things get weird so round them
+        self._selection._limits = self._limits
+
+    def __init__(
+        self,
+        selection: Sequence[float],
+        limits: Sequence[float],
+        axis: str = None,
+        parent: Graphic = None,
+        resizable: bool = True,
+        fill_color=(0, 0, 0.35),
+        edge_color=(0.8, 0.6, 0),
+        edge_thickness: float = 8,
+        vertex_color=(0.7, 0.4, 0),
+        vertex_thickness: float = 8,
+        arrow_keys_modifier: str = "Shift",
+        name: str = None,
+    ):
+        """
+        Create a RectangleSelector graphic which can be used to select a rectangular region of data.
+        Allows sub-selecting data from a ``Graphic`` or from multiple Graphics.
+
+        Parameters
+        ----------
+        selection: (float, float, float, float)
+            the initial selection of the rectangle, ``(x_min, x_max, y_min, y_max)``
+
+        limits: (float, float, float, float)
+            limits of the selector, ``(x_min, x_max, y_min, y_max)``
+
+        axis: str, default ``None``
+            Restrict the selector to the "x" or "y" axis.
+            If the selector is restricted to an axis you cannot change the selection along the other axis. For example,
+            if you set ``axis="x"``, then the ``y_min``, ``y_max`` values of the selection will stay constant.
```

Comment:
does this make sense here? I would imagine a `LinearRegionSelector` should be used if it needs to be restricted to an axis

Reply:
I agree, but I think you had it left over from the first implementation you did awhile ago...I will remove